shadows/framework/src/main/java/org/robolectric/shadows/ShadowParcelFileDescriptor.java

```diff
@@ -48,14 +51,8 @@ protected void __constructor__(ParcelFileDescriptor wrapped) {
 
   @Implementation
   protected static ParcelFileDescriptor open(File file, int mode) throws FileNotFoundException {
-    ParcelFileDescriptor pfd;
-    try {
-      Constructor<ParcelFileDescriptor> constructor =
-          ParcelFileDescriptor.class.getDeclaredConstructor(FileDescriptor.class);
-      pfd = constructor.newInstance(new FileDescriptor());
-    } catch (Exception e) {
-      throw new RuntimeException(e);
-    }
+    ParcelFileDescriptor pfd = new ParcelFileDescriptor(new FileDescriptor());
```

Comment:
Does this constructor public and open for all SDKs supported by Robolectric?

Reply:
This constructor is available in the interception of PFD. But it seems to be non-public SDK interface with `@hide` and `@UnsupportedAppUsage`.

https://cs.android.com/android/platform/superproject/+/master:frameworks/base/core/java/android/os/ParcelFileDescriptor.java;l=187;drc=master

Can we simplify the impl using it since it would not break the tests? I'm not sure about this.

Reply:
OK, I revert it. I don't think there's an API accessible restriction.  Non-public API calls only be rejected in Android runtime. But we can do more investigation and do it later.